pkg/controllers/management/clusterdeploy/clusterdeploy.go

```diff
@@ -253,7 +254,13 @@ func (cd *clusterDeploy) deployAgent(cluster *v3.Cluster) error {
 			output, err = kubectl.Delete([]byte(systemtemplate.AuthDaemonSet), kubeConfig)
 		}
 		if err != nil {
-			return cluster, errors.WithMessage(types.NewErrors(err, errors.New(string(output))), "kubectl delete failed")
+			logrus.Debugf("Output from kubectl delete kube-api-auth DaemonSet, output: %s, err: %v", string(output), err)
+			// Ignore if the resource does not exist and it returns 'daemonsets.apps "kube-api-auth" not found'
+			dsNotFoundError := "daemonsets.apps \"kube-api-auth\" not found"
+			if !strings.Contains(string(output), dsNotFoundError) {
```

Comment:
It looks like we could use the k8s `errors` package here `errors.IsNotFound()`
```suggestion
			if errors.IsNotFound(err) {
```

Reply:
i second this, LGTM if this is fixed

Reply:
The error returned from our `kubectl` package is `cmd.CombinedOutput()`: https://github.com/rancher/rancher/blob/master/pkg/kubectl/kubectl.go#L147

Reply:
oof